### PR TITLE
Cache setting

### PR DESCRIPTION
**Pull request by DavidMNorman**
# Checklist

- [ ] Bugfix
- [X] New feature
- [ ] Refactor

# Related Issue

With the addition of persistent queries requiring both the use of the wrapper and the router, we wanted to account for situations where developers wanted to use persistent queries without the use of a client-side cache.

Additionally, code needed cleaning and readme needed updating

# Solution

Added the option for the Obsidian Wrapper to contain a prop useCache. If a value of false is not provided, the client-side cache will default to being used. Additionally, included logic to prevent the cache from being used if caching is turned off.

removed console logs and made new searchSetting functionality clearer in commenting

### CODE REVIEW

File: ObsidianWrapper/ObsidianWrapper.jsx [hunk 1]
```diff
@@ -36,37 +42,19 @@ function ObsidianWrapper(props) {
     }
   });
 
-  const chromeExtensionId = 'apcpdmmbhhephobnmnllbklplpaoiemo';
-  // initialice cache in local storage
-  //window.localStorage.setItem('cache', JSON.stringify(cache));
-
-  let firstMessage = true;
```
Comment (exlxxn) [inline ObsidianWrapper/ObsidianWrapper.jsx:43]: I think this is for our DevTool?

File: ObsidianWrapper/ObsidianWrapper.jsx [hunk 2]
```diff
@@ -36,37 +42,19 @@ function ObsidianWrapper(props) {
...
+    // startDate is to find out when query was made, this data is passed to devtools
     const startTime = Date.now();
-    /*
-      chrome.runtime.sendMessage(chromeExtensionId, { query: query });
-      chrome.runtime.sendMessage(chromeExtensionId, {
-        cache: window.localStorage.getItem('cache'),
-      });
-    */
-
-    if(firstMessage){
```
Comment (exlxxn) [inline ObsidianWrapper/ObsidianWrapper.jsx:55]: This is also our for our dev tool? Unless if Joan changed it?

File: ObsidianWrapper/ObsidianWrapper.jsx [hunk 3]
```diff
@@ -93,32 +81,29 @@ function ObsidianWrapper(props) {
...
         console.log(
           "From cacheRead: Here's the response time on the front end: ",
           cacheHitResponseTime
         );
+
+        // FOR DEVTOOL - sends message to content.js with query metrics when query is a hit
         window.postMessage({
           type: 'query',
           time: cacheHitResponseTime,
-          name: startTime,
```
Comment (exlxxn) [inline ObsidianWrapper/ObsidianWrapper.jsx:109]: Why are we deleting this? I think this is also for our dev tool? Unless if we decided not to send a name value which maybe you did.

File: ObsidianWrapper/ObsidianWrapper.jsx [hunk 4]
```diff
@@ -138,26 +123,29 @@ function ObsidianWrapper(props) {
...
         console.log(
           "After the hunt: Here's the response time on the front end: ",
           cacheMissResponseTime
         );
+
+        // FOR DEVTOOL - sends message to content.js when query is a miss
         window.postMessage({
           type: 'query', 
           time: cacheMissResponseTime,
-          name: startTime,
```
Comment (exlxxn) [inline ObsidianWrapper/ObsidianWrapper.jsx:158]: Also why are we deleting this

File: ObsidianWrapper/ObsidianWrapper.jsx (hunk 3 above)
Comment (DavidMNorman) [reply ObsidianWrapper/ObsidianWrapper.jsx:109]: startTime is no longer being used by our Devtool. We use the time property instead.

File: ObsidianWrapper/ObsidianWrapper.jsx (hunk 2 above)
Comment (DavidMNorman) [reply ObsidianWrapper/ObsidianWrapper.jsx:55]: firstMessage was a leftover from the 5.0 system. Our devtool uses different messages.

File: ObsidianWrapper/ObsidianWrapper.jsx (hunk 4 above)
Comment (DavidMNorman) [reply ObsidianWrapper/ObsidianWrapper.jsx:158]: As with the other one, we've migrated the old start time functionality to simply use the time property's value instead.

File: ObsidianWrapper/ObsidianWrapper.jsx (hunk 1 above)
Comment (DavidMNorman) [reply ObsidianWrapper/ObsidianWrapper.jsx:43]: firstMessage was a leftover from the 5.0 system. Our devtool uses different messages.

**Review by exlxxn — APPROVED** [review]

**Pull request merged**